Source/App/EbAppInputy4m.c

```diff
@@ -308,20 +314,21 @@ EB_BOOL check_if_y4m(EbConfig_t *cfg) {
 
     /* Parse the header for the "YUV4MPEG2" string */
     headerReadLength = fread(buffer, YUV4MPEG2_IND_SIZE, 1, cfg->inputFile);
-    assert(headerReadLength == 1);
-    buffer[YUV4MPEG2_IND_SIZE] = 0;
-
-    validateAlphanumeric(buffer);
+    if (headerReadLength != 1) {
+        assert(feof(cfg->inputFile));
+        return EB_FALSE;
+    }
 
-    if (EB_STRCMP(buffer, "YUV4MPEG2") == 0) {
+    buffer[YUV4MPEG2_IND_SIZE] = 0;
+    if (validateAlphanumeric((char*)buffer) && EB_STRCMP(buffer, "YUV4MPEG2") == 0) {
         return EB_TRUE; /* YUV4MPEG2 file */
     }
     else {
         if (cfg->inputFile != stdin) {
             fseek(cfg->inputFile, 0, SEEK_SET);
         }
         else {
-            EB_STRNCPY(cfg->y4m_buf, buffer, YUV4MPEG2_IND_SIZE); /* TODO copy 9 bytes read to cfg->y4m_buf*/
+            EB_STRNCPY((char*)cfg->y4m_buf, buffer, YUV4MPEG2_IND_SIZE); /* TODO copy 9 bytes read to cfg->y4m_buf*/
         }
```

Comment:
y4m_buf is unsigned char type. I think it makes more sense that you change buffer to unsigned char.
BTW, raw video data is interpreted to be unsigned char or uint8_t type.